src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String MESSAGE_INDEX_ERROR = "Index outside the size of the list";
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private class Node<T> {
+        private T elementValue;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T elementValue, Node<T> next) {
+            this.previous = previous;
+            this.elementValue = elementValue;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addToTail(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (index == size) {
+            addToTail(value);
+        } else {
+            insertBefore(value, getNodeByIndex(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            addToTail(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexForGetAndSet(index);
+        return getNodeByIndex(index).elementValue;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T remove(int index) {
+        checkIndexForGetAndSet(index);
+        return unlinkNode(getNodeByIndex(index));
     }
 
     @Override
-    public T remove(int index) {
-        return null;
+    public boolean remove(T value) {
+        Node<T> nodeForRemove = head;
+        while (nodeForRemove != null) {
+            if (nodeForRemove.elementValue == null
```

Comment:
```suggestion
        while (nodeForRemove != null) {
            if (nodeForRemove.elementValue == value // we should check that they are both null
```

Reply:
modified to if ((nodeForRemove.elementValue == null && value == null)